sdk/resourcemanager/Azure.ResourceManager.Core/src/ApiVersions.cs

```diff
@@ -84,7 +85,15 @@ where method.GetParameters()[0].ParameterType == typeof(AzureResourceManagerClie
 
         private string LoadApiVersion(ResourceType resourceType, CancellationToken cancellationToken)
         {
-            var results = ProviderOperations.Get(resourceType.Namespace, null, cancellationToken);
+            Response<Provider> results;
+            try
+            {
+                results = ProviderOperations.Get(resourceType.Namespace, null, cancellationToken);
+            }
+            catch (RequestFailedException)
```

Comment:
Can we filter this with a "when" clause to not hide things like authentication etc?

Filter the exact error we want to capture

Reply:
This is the exact error, what other exception are we trying to catch? Are you saying we should catch anything, like this should just be Exception?